src/hotspot/share/memory/metaspace/counters.hpp

```diff
@@ -43,7 +44,7 @@ class AbstractCounter {
   T _c;
 
   // Only allow unsigned values for now
-  STATIC_ASSERT(IsSigned<T>::value == false);
+  STATIC_ASSERT(std::is_signed<T>::value == false);
```

Comment:
[pre-existing] Perhaps these should also be testing for `std::is_integral`,
since that is not implied by `std::is_signed`.  Although in the atomic case that
will fall out implicitly since we don't support atomic arithmetic on floating
point values.